app/src/main/java/org/oppia/app/parser/StringToRatioParser.kt

```diff
@@ -0,0 +1,74 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import androidx.annotation.StringRes
+import org.oppia.app.R
+import org.oppia.app.model.RatioExpression
+
+/** This class contains method that helps to parse string to ratio. */
+class StringToRatioParser {
+  private val invalidCharsRegex =
+    """^[\d\s/\:\.]+$""".toRegex()
+  private val invalidRatioRegex =
+    """^(\s)*(\d+((\s)*:(\s)*\d+)+)(\s)*$""".toRegex()
+
+  /**
+   * Returns a [RatioParsingError] for the specified text input if it's an invalid ratio, or
+   * [RatioParsingError.VALID] if no issues are found. Note that a valid ratio returned by this method is guaranteed
+   * to be parsed correctly by [parseRatioFromString].
+   *
+   * This method should only be used when a user tries submitting an answer. Real-time error detection should be done
+   * using [getRealTimeAnswerError], instead.
+   */
+  fun getSubmitTimeError(text: String, numberOfTerms: Int): RatioParsingError {
+    val ratio = parseRatio(text)
+    return if (ratio!!.serializedSize < numberOfTerms) {
```

Comment:
I don't think ``serializedSize`` is what you want--don't you want to check the number of terms? Are there tests for this parser?

Further, I suggest returning INVALID_FORMAT if the ratio fails to parse rather than throwing a NPE (otherwise you may end up with cases when an invalid fraction is inputted that gets through the real-time errors and triggers an NPE here due to parsing failure). I suggest adding a test for at least one of these cases if they exist.

Reply:
done